pysatModels/tests/test_utils_extract.py

```diff
@@ -14,321 +13,465 @@
-
         # Run everything successfully once
-        extract.extract_modelled_observations(*self.input_args,
-                                              **self.input_kwargs)
+        self.func(*self.input_args, **self.input_kwargs)
 
         # Run everything again, raising a value error
-        with pytest.raises(ValueError) as err:
-            extract.extract_modelled_observations(*self.input_args,
-                                                  **self.input_kwargs)
+        with caplog.at_level(logging.INFO, "pysatModels"):
+            with pytest.raises(ValueError) as err:
+                self.func(*self.input_args, **self.input_kwargs)
 
-            self.out = self.log_capture.getvalue()
-            assert self.out.find('model data already interpolated') >= 0
+        # Evaluate the logging messages and error messages
+        self.out = caplog.text
+        assert self.out.find('model data already interpolated') >= 0
 
         assert str(err.value.args[0]).find(
             'instrument object already contains all model data') >= 0
         return
 
-    def test_success_for_some_already_ran_data(self):
+    def test_success_for_some_already_ran_data(self, caplog):
         """Test the success for some model variables already extracted."""
 
         all_sel = list(self.input_kwargs['sel_name'])
-        self.input_kwargs['model_label'] = self.model_label
+        all_sel.append('dummy72')
 
-        # Run through twice
-        for i, selected in enumerate([all_sel[2:], all_sel]):
-            self.input_kwargs['sel_name'] = selected
-            self.out = extract.extract_modelled_observations(
-                *self.input_args, **self.input_kwargs)
-            lout = self.log_capture.getvalue()
+        self.model['dummy72'] = self.model['dummy2']
 
-        assert lout.find('model data already interpolated') >= 0
+        # Run through twice
+        with caplog.at_level(logging.INFO, logger='pysatModels'):
+            for i, selected in enumerate([all_sel[1:], all_sel]):
+                self.input_kwargs['sel_name'] = selected
+                self.input_kwargs['methods'] = ['linear'] * len(selected)
+                self.func(*self.input_args, **self.input_kwargs)
 
-        for label in all_sel:
-            if label not in self.input_args[3]:
-                # Test each of the extracted model data columns
-                tcol = "{:s}_{:s}".format(self.model_label, label)
-                assert tcol in self.inst.data.columns
-                assert (self.inst.data[self.input_args[2][0]].shape
-                        == self.inst.data[tcol].shape)
-                assert len(self.inst.data[tcol][
-                    ~np.isnan(self.inst.data[tcol])]) > 0
+            self.out = caplog.text
+            assert self.out.find('model data already interpolated') >= 0
 
+        # Evaluate output
+        self.input_kwargs['sel_name'] = pysat.utils.listify(all_sel[1])
```

Comment:
```suggestion
        self.input_kwargs['sel_name'] = pysat.utils.listify(all_sel[0])
```

Reply:
Had to switch to testing on `dummy2` as `dummy72` itself isn't consistent with expectations from eval_output.